client/blocks/login/index.jsx

```diff
@@ -69,22 +70,16 @@ class Login extends Component {
 	};
 
 	rebootAfterLogin = () => {
+		const { redirectTo } = this.props;
+
 		this.props.recordTracksEvent( 'calypso_login_success', {
 			two_factor_enabled: this.props.twoFactorEnabled
 		} );
 
-		const { redirectLocation } = this.props;
-
-		let newHref;
-
-		if ( redirectLocation && redirectLocation.match( /^(?!\/\/)[\/\-a-z0-9.]+$/i ) ) {
-			// only redirect to paths on the current domain
-			newHref = redirectLocation;
-		} else {
-			newHref = window.location.origin;
-		}
+		// Redirects to / if no redirect url is available
+		const url = redirectTo ? redirectTo : window.location.origin;
```

Comment:
Default values only work with [`undefined` properties](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Default_values_2). This wouldn't work in that case since `redirectTo` is always provided, even if it can be `null`.